# CS:GO Data Set by FERNANDOPY
Dataset from: https://www.kaggle.com/datasets/fernandopy/csgo-data-set?select=data (3 days ago)

# Objective
This dataset has 7 interconnected tables and I will use it for SQL practice. First, I need to set up the tables.

In [1]:
import pandas as pd
pd.set_option('display.max_columns', None)

# 1. CS:GO Players

In [2]:
csgo_player = pd.read_csv('players.csv')
csgo_player.head()

ID   NickName                        Name  Country
0  21468.0      hyped         Marcel 'hyped' Köhn  Germany
1  21139.0  pr1metapz      Luca 'pr1metapz' Voigt  Germany
2  20983.0       mrc9  Marc 'MRC9' Philipp\tJarke  Germany
3  21500.0      arrow          Luc 'ArroW' Oehmke  Germany
4  20169.0      skyye        Tom 'skyye' Hagedorn  Germany

In [3]:
csgo_player.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4399 entries, 0 to 4398
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   ID        4324 non-null   float64
 1   NickName  4324 non-null   object 
 2   Name      4398 non-null   object 
 3   Country   4399 non-null   object 
dtypes: float64(1), object(3)
memory usage: 137.6+ KB


Column 'ID' dtype are float, needs to be change it with int.\
There is null value.

In [4]:
csgo_player.isna().sum()

ID          75
NickName    75
Name         1
Country      0
dtype: int64

Column 'ID' and 'NickName' contain a lot of missing values. while the column 'Name' has only 1 missing value.\
Fill missing values in column 'ID' with 0 then change dtype to int.

In [5]:
csgo_player['ID'] = csgo_player['ID'].fillna(0)
csgo_player['ID'] = csgo_player['ID'].astype('int64')

In [6]:
csgo_player.duplicated().sum()

10

There are duplicate values, need to be deleted.

In [7]:
csgo_player.drop_duplicates(inplace=True)
csgo_player.reset_index(drop=True, inplace=True)
csgo_player.duplicated().sum()

0

In [8]:
csgo_player.info(verbose=False)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4389 entries, 0 to 4388
Columns: 4 entries, ID to Country
dtypes: int64(1), object(3)
memory usage: 137.3+ KB


#### CS:GO Players Results
Fill missing values in column 'ID' with 0.\
Change dtype column 'ID' from float to int.\
Drop 10 duplicate rows:
- Before 4399 
- After 4389

Extract CS:GO Players to Excel.

In [9]:
csgo_player.to_excel(('csgo_player.xlsx'))

# 2. CS:GO Teams

In [10]:
csgo_team = pd.read_csv('teams.csv')
csgo_team.head()

ID         Nick         Name  Country
0  10254  big-academy  BIG Academy  Germany
1  10831     entropiq     Entropiq   Europe
2  11883     9-pandas     9 Pandas   Russia
3  11283      falcons      Falcons   Europe
4  10621         1win         1WIN   Russia

In [11]:
csgo_team.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 735 entries, 0 to 734
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   ID       735 non-null    int64 
 1   Nick     735 non-null    object
 2   Name     735 non-null    object
 3   Country  735 non-null    object
dtypes: int64(1), object(3)
memory usage: 23.1+ KB


There is no missing value and column 'ID' dtype is already int.

In [12]:
csgo_team.duplicated().sum()

1

There is 1 duplicate value, need to be deleted.

In [13]:
csgo_team.drop_duplicates(inplace=True)
csgo_team.reset_index(drop=True, inplace=True)
csgo_team.duplicated().sum()

0

In [14]:
csgo_team.info(verbose=False)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 734 entries, 0 to 733
Columns: 4 entries, ID to Country
dtypes: int64(1), object(3)
memory usage: 23.1+ KB


#### CS:GO Teams Results
Drop 1 duplicate rows:
- before 735 
- after 734 

Extract CS:GO Teams to Excel.

In [15]:
csgo_team.to_excel(('csgo_team.xlsx'))

# 3. CS:GO Match Ids

In [16]:
csgo_match_id = pd.read_csv('matchIds.csv')
csgo_match_id.head()

ID                                             Tittle
0  2366986  2366986/big-academy-vs-entropiq-esea-advanced-...
1  2366987  2366987/los-kogutos-vs-7am-esea-advanced-seaso...
2  2366947  2366947/9-pandas-vs-falcons-cct-north-europe-s...
3  2366807      2366807/1win-vs-mibr-cct-2023-online-finals-3
4  2366946  2366946/ecstatic-vs-aurora-young-blud-cct-nort...

Column 'Tittle' needs to be fixed.

In [17]:
csgo_match_id.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 78236 entries, 0 to 78235
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   ID      78236 non-null  int64 
 1   Tittle  78236 non-null  object
dtypes: int64(1), object(1)
memory usage: 1.2+ MB


There is no missing value and column 'ID' dtype is already int.

In [18]:
csgo_match_id.duplicated().sum()

0

There is no duplicated value.

In [19]:
csgo_match_id['Tittle'] = csgo_match_id['Tittle'].str.replace('-',' ')
csgo_match_id['Tittle'] = csgo_match_id['Tittle'].str.split('/').str[1]
csgo_match_id['Tittle']

0        big academy vs entropiq esea advanced season 4...
1        los kogutos vs 7am esea advanced season 46 europe
2            9 pandas vs falcons cct north europe series 8
3                    1win vs mibr cct 2023 online finals 3
4        ecstatic vs aurora young blud cct north europe...
                               ...                        
78231    ninjas in pyjamas vs lemondogs steelseries go ...
78232             lemondogs vs e srael steelseries go 2012
78233         ninjas in pyjamas vs vby steelseries go 2012
78234                 lemondogs vs vby steelseries go 2012
78235     ninjas in pyjamas vs e srael steelseries go 2012
Name: Tittle, Length: 78236, dtype: object

#### CS:GO Match Ids Results
Fix column 'Tittle'.

Extract CS:GO Match Ids to Excel.

In [20]:
csgo_match_id.to_excel(('csgo_match_id.xlsx'))

# 4. CS:GO Match Map

In [21]:
csgo_match_map = pd.read_csv('matchMap.csv')
csgo_match_map.head()

MatchID   MapID   MapName
0  2366986/big-academy-vs-entropiq-esea-advanced-...  163463   Inferno
1  2366986/big-academy-vs-entropiq-esea-advanced-...  163467    Anubis
2  2366986/big-academy-vs-entropiq-esea-advanced-...  163477  Overpass
3  2366987/los-kogutos-vs-7am-esea-advanced-seaso...  163470   Ancient
4  2366987/los-kogutos-vs-7am-esea-advanced-seaso...  163478    Anubis

Column 'MatchID' needs to be fixed.

In [22]:
csgo_match_map.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1117 entries, 0 to 1116
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   MatchID  1117 non-null   object
 1   MapID    1117 non-null   int64 
 2   MapName  1117 non-null   object
dtypes: int64(1), object(2)
memory usage: 26.3+ KB


Column 'MatchID' dtype are object, needs to be change it with int.\
There is no missing value.

In [23]:
csgo_match_map.duplicated().sum()

0

There is no duplicated value.

In [24]:
csgo_match_map['MatchID'] = csgo_match_map['MatchID'].str.split('/').str[0]
csgo_match_map['MatchID'] = csgo_match_map['MatchID'].astype('int64')
csgo_match_map['MatchID']

0       2366986
1       2366986
2       2366986
3       2366987
4       2366987
         ...   
1112    2366381
1113    2366381
1114    2366381
1115    2366340
1116    2366340
Name: MatchID, Length: 1117, dtype: int64

#### CS:GO Match Map Results
Fix column 'MatchID' and change dtype from object to int.

Extract CS:GO Match Map to Excel.

In [25]:
csgo_match_map.to_excel(('csgo_match_map.xlsx'))

# 5. CS:GO Match Overview

In [26]:
csgo_match_overview = pd.read_csv('matchOverview.csv')
csgo_match_overview.head()

MatchID   MapName  Team1  Team2  Team1_Pick  Team1_Win  Team1_Final_Score  \
0  2366986   Inferno  10254  10831       False      False                  9   
1  2366986    Anubis  10254  10831        True       True                 16   
2  2366986  Overpass  10254  10831       False       True                 16   
3  2366947  Overpass  11883  11283        True      False                  8   
4  2366947    Mirage  11883  11283       False      False                 12   

   Team2_Final_Score  Halfs_Played Team1_Start_Side  Team1_Half1_Score  \
0                 16             2               CT                  4   
1                 12             2               TR                  8   
2                 14             2               CT                  8   
3                 16             2               TR                  6   
4                 16             2               CT                  9   

   Team2_Half1_Score  Team1_Half2_Score  Team2_Half2_Score  Team1_OT_Score  \
0                 11                  5                  5               0   
1                  7                  8                  5               0   
2                  7                  8                  7               0   
3                  9                  2                  7               0   
4                  6                  3                 10               0   

   Team2_OT_Score  
0               0  
1               0  
2               0  
3               0  
4               0

Column 'MapName' need to change to 'MapID' from CS:GO Match Map.

In [27]:
csgo_match_overview.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20016 entries, 0 to 20015
Data columns (total 16 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   MatchID            20016 non-null  int64 
 1   MapName            20016 non-null  object
 2   Team1              20016 non-null  int64 
 3   Team2              20016 non-null  int64 
 4   Team1_Pick         20016 non-null  bool  
 5   Team1_Win          20016 non-null  bool  
 6   Team1_Final_Score  20016 non-null  int64 
 7   Team2_Final_Score  20016 non-null  int64 
 8   Halfs_Played       20016 non-null  int64 
 9   Team1_Start_Side   20016 non-null  object
 10  Team1_Half1_Score  20016 non-null  int64 
 11  Team2_Half1_Score  20016 non-null  int64 
 12  Team1_Half2_Score  20016 non-null  int64 
 13  Team2_Half2_Score  20016 non-null  int64 
 14  Team1_OT_Score     20016 non-null  int64 
 15  Team2_OT_Score     20016 non-null  int64 
dtypes: bool(2), int64(12), object(2)
memory 

There is no missing value and column 'MatchID', 'Team1', 'Team2' dtype is already int.

In [28]:
csgo_match_overview.duplicated().sum()

0

There is no duplicated value.

Adding column 'MapID' from CS:GO Match Map to CS:GO Match Overview by merging based on column 'MatchID' and 'MapName'.

In [29]:
csgo_moverview_merge = csgo_match_overview.merge(csgo_match_map,left_on=['MatchID', 'MapName'], right_on=['MatchID', 'MapName'])

Placing column 'MapID' where column 'MapName' position is located by reindex the column.\
Also drop column 'MapName'.

In [30]:
csgo_moverview_merge.columns

Index(['MatchID', 'MapName', 'Team1', 'Team2', 'Team1_Pick', 'Team1_Win',
       'Team1_Final_Score', 'Team2_Final_Score', 'Halfs_Played',
       'Team1_Start_Side', 'Team1_Half1_Score', 'Team2_Half1_Score',
       'Team1_Half2_Score', 'Team2_Half2_Score', 'Team1_OT_Score',
       'Team2_OT_Score', 'MapID'],
      dtype='object')

In [31]:
csgo_moverview_merge = csgo_moverview_merge.reindex(['MatchID', 'MapName', 'MapID', 'Team1', 'Team2', 'Team1_Pick', 'Team1_Win',
                                                     'Team1_Final_Score', 'Team2_Final_Score', 'Halfs_Played',
                                                     'Team1_Start_Side', 'Team1_Half1_Score', 'Team2_Half1_Score',
                                                     'Team1_Half2_Score', 'Team2_Half2_Score', 'Team1_OT_Score',
                                                     'Team2_OT_Score'], axis=1)
csgo_moverview_merge.drop(columns='MapName', axis=1, inplace=True)
csgo_moverview_merge

MatchID   MapID  Team1  Team2  Team1_Pick  Team1_Win  Team1_Final_Score  \
0     2366986  163463  10254  10831       False      False                  9   
1     2366986  163467  10254  10831        True       True                 16   
2     2366986  163477  10254  10831       False       True                 16   
3     2366947  163464  11883  11283        True      False                  8   
4     2366947  163474  11883  11283       False      False                 12   
...       ...     ...    ...    ...         ...        ...                ...   
1110  2366381  161913   3044  11306       False       True                 16   
1111  2366381  161920   3044  11306        True      False                 16   
1112  2366381  161922   3044  11306       False       True                 22   
1113  2366340  161915  10399   9799       False       True                 16   
1114  2366340  161919  10399   9799        True       True                 16   

      Team2_Final_Score  Halfs_Played Team1_Start_Side  Team1_Half1_Score  \
0                    16             2               CT                  4   
1                    12             2               TR                  8   
2                    14             2               CT                  8   
3                    16             2               TR                  6   
4                    16             2               CT                  9   
...                 ...           ...              ...                ...   
1110                 12             2               CT                  9   
1111                 19             2               TR                  7   
1112                 18             2               TR                  6   
1113                  3             2               CT                 13   
1114                  7             2               TR                  9   

      Team2_Half1_Score  Team1_Half2_Score  Team2_Half2_Score  Team1_OT_Score  \
0                    11                  5                  5               0   
1                     7                  8                  5               0   
2                     7                  8                  7               0   
3                     9                  2                  7               0   
4                     6                  3                 10               0   
...                 ...                ...                ...             ...   
1110                  6                  7                  6               0   
1111                  8                  8                  7               1   
1112                  9                  9                  6               7   
1113                  2                  3                  1               0   
1114                  6                  7                  1               0   

      Team2_OT_Score  
0                  0  
1                  0  
2                  0  
3                  0  
4                  0  
...              ...  
1110               0  
1111               4  
1112               3  
1113               0  
1114               0  

[1115 rows x 16 columns]

In [32]:
csgomap_map_unique = csgo_match_map['MapName'].nunique()
csgomap_mid_unique = csgo_match_map['MatchID'].nunique()

csgo_map_print = "CS:GO Match Map have "+str(csgomap_map_unique)+" unique 'MapName' values, and "+str(csgomap_mid_unique)+" unique 'MatchID' values"

csgoover_map_unique = csgo_match_overview['MapName'].nunique()
csgoover_mid_unique = csgo_match_overview['MatchID'].nunique()

csgo_match_overrow = len(csgo_match_overview)
csgo_mover_mergerow = len(csgo_moverview_merge)
lost_merge_row1 = (csgo_match_overrow-csgo_mover_mergerow)/csgo_match_overrow*100

print(csgo_map_print)
print("CS:GO Match Overview have "+str(csgoover_map_unique)+" unique 'MapName' values, and "+
      str(csgoover_mid_unique)+" unique 'MatchID' values")
print("Before, CS:GO Match Overview has "+str(csgo_match_overrow)+" rows. \
      \nCS:GO Match Overview now have "+str(csgo_mover_mergerow)+" rows.\
      \nCS:GO Match Overview lost "+str(round(lost_merge_row1, 2))+" rows after merge it with CS:GO Match Map based on column 'MatchID' and 'MapName'.")

CS:GO Match Map have 7 unique 'MapName' values, and 490 unique 'MatchID' values
CS:GO Match Overview have 10 unique 'MapName' values, and 9346 unique 'MatchID' values
Before, CS:GO Match Overview has 20016 rows.       
CS:GO Match Overview now have 1115 rows.      
CS:GO Match Overview lost 94.43 rows after merge it with CS:GO Match Map based on column 'MatchID' and 'MapName'.


#### CS:GO Match Overview Results
Merge CS:GO Match Overview with CS:GO Match Map based on column 'MatchID' and 'MapName' on both side
- Before 20016 rows
- After 1115 rows
- Lost 94.43 rows

Extract CS:GO Match Overview to Excel.

In [33]:
csgo_moverview_merge.to_excel(('csgo_moverview_merge.xlsx'))

# 6. CS:GO Match Infos

In [34]:
csgo_match_info = pd.read_csv('matchInfos.csv')
csgo_match_info.head()

PlayerID  TeamID  TeamRank  MatchID  MapName  Total_Kills  Total_Deaths  \
0     21468   10254        80  2366986  Inferno         14.0          18.0   
1     21139   10254        80  2366986  Inferno         14.0          19.0   
2     20983   10254        80  2366986  Inferno         10.0          20.0   
3     21500   10254        80  2366986  Inferno         15.0          19.0   
4     20169   10254        80  2366986  Inferno         20.0          17.0   

   Total_Score  Total_ADR  Total_KAST  Total_Rating  CT_Kills  CT_Deaths  \
0         -4.0       55.2        0.52          0.73       6.0       11.0   
1         -5.0       66.8        0.64          0.81       8.0       13.0   
2        -10.0       48.4        0.40          0.64       0.0       14.0   
3         -4.0       80.6        0.68          1.03      11.0       11.0   
4          3.0       81.8        0.60          1.05      16.0       11.0   

   CT_Score  CT_ADR  CT_KAST  CT_Rating  TR_Kills  TR_Deaths  TR_Score  \
0      -5.0    42.0     0.46       0.42       8.0        7.0       1.0   
1      -5.0    63.3     0.66       0.68       6.0        6.0       0.0   
2     -14.0    15.3     0.13       0.03      10.0        6.0       4.0   
3       0.0    97.9     0.60       1.09       4.0        8.0      -4.0   
4       5.0   114.6     0.66       1.34       4.0        6.0      -2.0   

   TR_ADR  TR_KAST  TR_Rating  
0    74.9      0.6       1.19  
1    72.2      0.6       0.99  
2    98.1      0.8       1.54  
3    54.5      0.8       0.94  
4    32.7      0.5       0.60

Column 'MapName' need to change to 'MapID' from CS:GO Match Map.

In [35]:
csgo_match_info.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 201790 entries, 0 to 201789
Data columns (total 23 columns):
 #   Column        Non-Null Count   Dtype  
---  ------        --------------   -----  
 0   PlayerID      201790 non-null  int64  
 1   TeamID        201790 non-null  int64  
 2   TeamRank      201790 non-null  int64  
 3   MatchID       201790 non-null  int64  
 4   MapName       201790 non-null  object 
 5   Total_Kills   201240 non-null  float64
 6   Total_Deaths  201240 non-null  float64
 7   Total_Score   201240 non-null  float64
 8   Total_ADR     201240 non-null  float64
 9   Total_KAST    201240 non-null  float64
 10  Total_Rating  201240 non-null  float64
 11  CT_Kills      201240 non-null  float64
 12  CT_Deaths     201240 non-null  float64
 13  CT_Score      201240 non-null  float64
 14  CT_ADR        201240 non-null  float64
 15  CT_KAST       201240 non-null  float64
 16  CT_Rating     201240 non-null  float64
 17  TR_Kills      201240 non-null  float64
 18  TR_D

Column 'PlayerID', 'TeamID', 'MatchID' dtype is already int.\
There is null value.\
There are incorrect dtype for column:
- Total_Kills, from float to int.
- Total_Deaths, from float to int.
- Total_Score, from float to int.
- CT_Kills, from float to int.
- CT_Deaths, from float to int.
- CT_Score, from float to int.
- TR_Kills, from float to int.
- TR_Deaths, from float to int.
- TR_Score, from float to int.

In [36]:
csgo_match_info.duplicated().sum()

0

There is no duplicated value.

Adding column 'MapID' from CS:GO Match Map to CS:GO Match Infos by merging based on column 'MatchID' and 'MapName'.

In [37]:
csgo_minfo_merge = csgo_match_info.merge(csgo_match_map,left_on=['MatchID', 'MapName'], right_on=['MatchID', 'MapName'])

Placing column 'MapID' where column 'MapName' position is located by reindex the column.\
Also drop column 'MapName'.

In [38]:
csgo_minfo_merge.columns

Index(['PlayerID', 'TeamID', 'TeamRank', 'MatchID', 'MapName', 'Total_Kills',
       'Total_Deaths', 'Total_Score', 'Total_ADR', 'Total_KAST',
       'Total_Rating', 'CT_Kills', 'CT_Deaths', 'CT_Score', 'CT_ADR',
       'CT_KAST', 'CT_Rating', 'TR_Kills', 'TR_Deaths', 'TR_Score', 'TR_ADR',
       'TR_KAST', 'TR_Rating', 'MapID'],
      dtype='object')

In [39]:
csgo_minfo_merge = csgo_minfo_merge.reindex(['PlayerID', 'TeamID', 'TeamRank', 'MatchID', 'MapID', 'MapName', 'Total_Kills',
                                             'Total_Deaths', 'Total_Score', 'Total_ADR', 'Total_KAST',
                                             'Total_Rating', 'CT_Kills', 'CT_Deaths', 'CT_Score', 'CT_ADR',
                                             'CT_KAST', 'CT_Rating', 'TR_Kills', 'TR_Deaths', 'TR_Score', 'TR_ADR',
                                             'TR_KAST', 'TR_Rating'], axis=1)
csgo_minfo_merge.drop(columns='MapName', axis=1, inplace=True)
csgo_minfo_merge

PlayerID  TeamID  TeamRank  MatchID   MapID  Total_Kills  Total_Deaths  \
0         21468   10254        80  2366986  163463         14.0          18.0   
1         21139   10254        80  2366986  163463         14.0          19.0   
2         20983   10254        80  2366986  163463         10.0          20.0   
3         21500   10254        80  2366986  163463         15.0          19.0   
4         20169   10254        80  2366986  163463         20.0          17.0   
...         ...     ...       ...      ...     ...          ...           ...   
11145      7253    9799       121  2366340  161919         10.0          21.0   
11146     12113    9799       121  2366340  161919         12.0          20.0   
11147      8711    9799       121  2366340  161919         17.0          19.0   
11148     16318    9799       121  2366340  161919         14.0          16.0   
11149     20215    9799       121  2366340  161919         13.0          16.0   

       Total_Score  Total_ADR  Total_KAST  Total_Rating  CT_Kills  CT_Deaths  \
0             -4.0       55.2        0.52          0.73       6.0       11.0   
1             -5.0       66.8        0.64          0.81       8.0       13.0   
2            -10.0       48.4        0.40          0.64       0.0       14.0   
3             -4.0       80.6        0.68          1.03      11.0       11.0   
4              3.0       81.8        0.60          1.05      16.0       11.0   
...            ...        ...         ...           ...       ...        ...   
11145        -11.0       57.7        0.43          0.48       5.0       14.0   
11146         -8.0       63.4        0.43          0.59      12.0       13.0   
11147         -2.0       85.7        0.52          0.94      14.0       11.0   
11148         -2.0       70.3        0.69          0.97      11.0        9.0   
11149         -3.0       59.3        0.60          0.82       7.0        9.0   

       CT_Score  CT_ADR  CT_KAST  CT_Rating  TR_Kills  TR_Deaths  TR_Score  \
0          -5.0    42.0     0.46       0.42       8.0        7.0       1.0   
1          -5.0    63.3     0.66       0.68       6.0        6.0       0.0   
2         -14.0    15.3     0.13       0.03      10.0        6.0       4.0   
3           0.0    97.9     0.60       1.09       4.0        8.0      -4.0   
4           5.0   114.6     0.66       1.34       4.0        6.0      -2.0   
...         ...     ...      ...        ...       ...        ...       ...   
11145      -9.0    48.3     0.40       0.38       5.0        7.0      -2.0   
11146      -1.0    90.5     0.60       0.87       0.0        7.0      -7.0   
11147       3.0    98.3     0.60       1.15       3.0        8.0      -5.0   
11148       2.0    81.3     0.73       1.13       3.0        7.0      -4.0   
11149      -2.0    47.4     0.60       0.68       6.0        7.0      -1.0   

       TR_ADR  TR_KAST  TR_Rating  
0        74.9     0.60       1.19  
1        72.2     0.60       0.99  
2        98.1     0.80       1.54  
3        54.5     0.80       0.94  
4        32.7     0.50       0.60  
...       ...      ...        ...  
11145    75.2     0.50       0.67  
11146    12.6     0.12       0.08  
11147    62.0     0.37       0.56  
11148    49.6     0.62       0.67  
11149    81.5     0.62       1.08  

[11150 rows x 23 columns]

In [40]:
csgoinfo_map_unique = csgo_match_info['MapName'].nunique()
csgoinfo_mid_unique = csgo_match_info['MatchID'].nunique()

csgo_match_inforow = len(csgo_match_info)
csgo_minfo_mergerow = len(csgo_minfo_merge)
lost_merge_row2 = (csgo_match_inforow-csgo_minfo_mergerow)/csgo_match_inforow*100

print(csgo_map_print)
print("CS:GO Match Infos have "+str(csgoinfo_map_unique)+" unique 'MapName' values, and "+
      str(csgoinfo_mid_unique)+" unique 'MatchID' values")
print("Before, CS:GO Match Infos has "+str(csgo_match_inforow)+" rows. \
      \nCS:GO Match Infos now have "+str(csgo_minfo_mergerow)+" rows.\
      \nCS:GO Match Infos lost "+str(round(lost_merge_row2, 2))+" rows after merge it with CS:GO Match Map based on column 'MatchID' and 'MapName'.")

CS:GO Match Map have 7 unique 'MapName' values, and 490 unique 'MatchID' values
CS:GO Match Infos have 9 unique 'MapName' values, and 9420 unique 'MatchID' values
Before, CS:GO Match Infos has 201790 rows.       
CS:GO Match Infos now have 11150 rows.      
CS:GO Match Infos lost 94.47 rows after merge it with CS:GO Match Map based on column 'MatchID' and 'MapName'.


In [41]:
csgo_minfo_merge.isna().sum()

PlayerID         0
TeamID           0
TeamRank         0
MatchID          0
MapID            0
Total_Kills     26
Total_Deaths    26
Total_Score     26
Total_ADR       26
Total_KAST      26
Total_Rating    26
CT_Kills        26
CT_Deaths       26
CT_Score        26
CT_ADR          26
CT_KAST         26
CT_Rating       26
TR_Kills        26
TR_Deaths       26
TR_Score        26
TR_ADR          26
TR_KAST         26
TR_Rating       26
dtype: int64

For the last 18 column have missing value and need to be deleted.

In [42]:
csgo_minfo_merge.dropna(inplace=True)
csgo_minfo_merge.reset_index(drop=True, inplace=True)
csgo_minfo_merge.shape[0]

11124

Change dtype in several column from float to int.

In [43]:
csgo_minfo_merge['Total_Kills'] = csgo_minfo_merge['Total_Kills'].astype('int64')
csgo_minfo_merge['Total_Deaths'] = csgo_minfo_merge['Total_Deaths'].astype('int64')
csgo_minfo_merge['Total_Score'] = csgo_minfo_merge['Total_Score'].astype('int64')
csgo_minfo_merge['CT_Kills'] = csgo_minfo_merge['CT_Kills'].astype('int64')
csgo_minfo_merge['CT_Deaths'] = csgo_minfo_merge['CT_Deaths'].astype('int64')
csgo_minfo_merge['CT_Score'] = csgo_minfo_merge['CT_Score'].astype('int64')
csgo_minfo_merge['TR_Kills'] = csgo_minfo_merge['TR_Kills'].astype('int64')
csgo_minfo_merge['TR_Deaths'] = csgo_minfo_merge['TR_Deaths'].astype('int64')
csgo_minfo_merge['TR_Score'] = csgo_minfo_merge['TR_Score'].astype('int64')

#### CS:GO Match Infos Results
Merge CS:GO Match Infos with CS:GO Match Map based on column 'MatchID' and 'MapName' on both side.
- Before 201790 rows
- After 11150 rows
- Lost 94.47 rows

Drop 26 duplicate rows:
- Before 11150 rows
- After 11124 rows

Change column dtype for column:
- Total_Kills, from float to int.
- Total_Deaths, from float to int.
- Total_Score, from float to int.
- CT_Kills, from float to int.
- CT_Deaths, from float to int.
- CT_Score, from float to int.
- TR_Kills, from float to int.
- TR_Deaths, from float to int.
- TR_Score, from float to int.

Extract CS:GO Match Infos to Excel.

In [44]:
csgo_minfo_merge.to_excel(('csgo_minfo_merge.xlsx'))

# 7. CS:GO Match Economy

In [45]:
csgo_match_economy = pd.read_csv('matchEconomy.csv')
csgo_match_economy.head()

MatchID  TeamID   MapName  Eco_Rounds  Eco_Rounds_Wons  Semi_Eco_Rounds  \
0  2366986   10254   Inferno           1                0                4   
1  2366986   10831   Inferno           1                0                0   
2  2366986   10254    Anubis           0                0                2   
3  2366986   10831    Anubis           1                0                3   
4  2366986   10254  Overpass           2                0                1   

   Semi_Eco_Rounds_Wons  Semi_Buy_Rounds  Semi_Buy_Rounds_Wons  \
0                     1                3                     1   
1                     0                3                     1   
2                     0                3                     1   
3                     2                5                     2   
4                     0                2                     0   

   Full_Buy_Rounds  Full_Buy_Rounds_Wons  Round1_Spent  Round1_Result  \
0               15                     6          4300          False   
1               19                    14          4200           True   
2               21                    13          4200           True   
3               17                     8          4250          False   
4               23                    16          4200          False   

   Round2_Spent  Round2_Result  Round3_Spent  Round3_Result  Round4_Spent  \
0          9400          False          1000          False         27950   
1         18750           True         22350           True         22200   
2         19750          False          9550          False         15000   
3          9650           True         22650           True         31650   
4         11900          False          1400          False         24350   

   Round4_Result  Round5_Spent  Round5_Result  Round6_Spent  Round6_Result  \
0          False         11550          False         29250          False   
1           True         28450           True         26650           True   
2          False          6800          False         24600           True   
3           True         31050           True         29800          False   
4           True         28150          False         24950           True   

   Round7_Spent  Round7_Result  Round8_Spent  Round8_Result  Round9_Spent  \
0          9100           True         29750           True         28000   
1         27950          False         26350          False         26650   
2         26150           True         25650           True         26450   
3         15050          False          7100          False         22750   
4         28750           True         29200           True         21200   

   Round9_Result  Round10_Spent  Round10_Result  Round11_Spent  \
0          False          22650           False          10600   
1           True          25450            True          27150   
2           True          25850            True          26850   
3          False          12400           False          26000   
4           True          30950            True          27950   

   Round11_Result  Round12_Spent  Round12_Result  Round13_Spent  \
0           False          26650           False           9800   
1            True          27450            True          27450   
2            True          27650           False          26150   
3           False           9300            True          29400   
4            True          30450           False          28850   

   Round13_Result  Round14_Spent  Round14_Result  Round15_Spent  \
0           False          26550            True          30750   
1            True          26950           False          26950   
2            True          26450           False          26350   
3           False          30300            True          32150   
4           False          21950            True          30000   

   Round15_Result  Round16_Spent  Round16_Result  Round17_Spent  \
0            True           4200 

Column 'MapName' need to change to 'MapID' from CS:GO Match Map.

In [46]:
csgo_match_economy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 39914 entries, 0 to 39913
Data columns (total 71 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   MatchID               39914 non-null  int64  
 1   TeamID                39914 non-null  int64  
 2   MapName               39914 non-null  object 
 3   Eco_Rounds            39914 non-null  int64  
 4   Eco_Rounds_Wons       39914 non-null  int64  
 5   Semi_Eco_Rounds       39914 non-null  int64  
 6   Semi_Eco_Rounds_Wons  39914 non-null  int64  
 7   Semi_Buy_Rounds       39914 non-null  int64  
 8   Semi_Buy_Rounds_Wons  39914 non-null  int64  
 9   Full_Buy_Rounds       39914 non-null  int64  
 10  Full_Buy_Rounds_Wons  39914 non-null  int64  
 11  Round1_Spent          39914 non-null  int64  
 12  Round1_Result         39914 non-null  bool   
 13  Round2_Spent          39914 non-null  int64  
 14  Round2_Result         39914 non-null  bool   
 15  Round3_Spent       

Column 'MatchID', and 'TeamID' dtype is already int.\
There are null value start from column 'Round17_Spent' to end.\
Also column dtype is float from column 'Round17_Spent'to end.\
Thats because most CS:GO Match end up to round 16, CS:GO Match end up to round 17 or even round 30 are very unusual.\
No need to change dtype

In [47]:
csgo_match_economy.duplicated().sum()

0

There is no duplicated value.

Adding column 'MapID' from CS:GO Match Map to CS:GO Match Economy by merging based on column 'MatchID' and 'MapName'.

In [48]:
csgo_meco_merge = csgo_match_economy.merge(csgo_match_map,left_on=['MatchID', 'MapName'], right_on=['MatchID', 'MapName'])

Placing column 'MapID' where column 'MapName' position is located by reindex the column.\
Also drop column 'MapName'.

In [49]:
csgo_meco_merge.columns

Index(['MatchID', 'TeamID', 'MapName', 'Eco_Rounds', 'Eco_Rounds_Wons',
       'Semi_Eco_Rounds', 'Semi_Eco_Rounds_Wons', 'Semi_Buy_Rounds',
       'Semi_Buy_Rounds_Wons', 'Full_Buy_Rounds', 'Full_Buy_Rounds_Wons',
       'Round1_Spent', 'Round1_Result', 'Round2_Spent', 'Round2_Result',
       'Round3_Spent', 'Round3_Result', 'Round4_Spent', 'Round4_Result',
       'Round5_Spent', 'Round5_Result', 'Round6_Spent', 'Round6_Result',
       'Round7_Spent', 'Round7_Result', 'Round8_Spent', 'Round8_Result',
       'Round9_Spent', 'Round9_Result', 'Round10_Spent', 'Round10_Result',
       'Round11_Spent', 'Round11_Result', 'Round12_Spent', 'Round12_Result',
       'Round13_Spent', 'Round13_Result', 'Round14_Spent', 'Round14_Result',
       'Round15_Spent', 'Round15_Result', 'Round16_Spent', 'Round16_Result',
       'Round17_Spent', 'Round17_Result', 'Round18_Spent', 'Round18_Result',
       'Round19_Spent', 'Round19_Result', 'Round20_Spent', 'Round20_Result',
       'Round21_Spent', 'Round21_

In [50]:
csgo_meco_merge = csgo_meco_merge.reindex(['MatchID', 'TeamID', 'MapName', 'MapID', 'Eco_Rounds', 'Eco_Rounds_Wons',
                                            'Semi_Eco_Rounds', 'Semi_Eco_Rounds_Wons', 'Semi_Buy_Rounds',
                                            'Semi_Buy_Rounds_Wons', 'Full_Buy_Rounds', 'Full_Buy_Rounds_Wons',
                                            'Round1_Spent', 'Round1_Result', 'Round2_Spent', 'Round2_Result',
                                            'Round3_Spent', 'Round3_Result', 'Round4_Spent', 'Round4_Result',
                                            'Round5_Spent', 'Round5_Result', 'Round6_Spent', 'Round6_Result',
                                            'Round7_Spent', 'Round7_Result', 'Round8_Spent', 'Round8_Result',
                                            'Round9_Spent', 'Round9_Result', 'Round10_Spent', 'Round10_Result',
                                            'Round11_Spent', 'Round11_Result', 'Round12_Spent', 'Round12_Result',
                                            'Round13_Spent', 'Round13_Result', 'Round14_Spent', 'Round14_Result',
                                            'Round15_Spent', 'Round15_Result', 'Round16_Spent', 'Round16_Result',
                                            'Round17_Spent', 'Round17_Result', 'Round18_Spent', 'Round18_Result',
                                            'Round19_Spent', 'Round19_Result', 'Round20_Spent', 'Round20_Result',
                                            'Round21_Spent', 'Round21_Result', 'Round22_Spent', 'Round22_Result',
                                            'Round23_Spent', 'Round23_Result', 'Round24_Spent', 'Round24_Result',
                                            'Round25_Spent', 'Round25_Result', 'Round26_Spent', 'Round26_Result',
                                            'Round27_Spent', 'Round27_Result', 'Round28_Spent', 'Round28_Result',
                                            'Round29_Spent', 'Round29_Result', 'Round30_Spent', 'Round30_Result'], axis=1)
csgo_meco_merge.drop(columns='MapName', axis=1, inplace=True)
csgo_meco_merge

MatchID  TeamID   MapID  Eco_Rounds  Eco_Rounds_Wons  Semi_Eco_Rounds  \
0     2366986   10254  163463           1                0                4   
1     2366986   10831  163463           1                0                0   
2     2366986   10254  163467           0                0                2   
3     2366986   10831  163467           1                0                3   
4     2366986   10254  163477           2                0                1   
...       ...     ...     ...         ...              ...              ...   
2223  2366340    9799  161915           1                0                5   
2224  2366340   10399  161919           1                0                0   
2225  2366340    9799  161919           2                0                3   
2226  2366774   12443  163197           3                0                1   
2227  2366774   12190  163197           3                1                1   

      Semi_Eco_Rounds_Wons  Semi_Buy_Rounds  Semi_Buy_Rounds_Wons  \
0                        1                3                     1   
1                        0                3                     1   
2                        0                3                     1   
3                        2                5                     2   
4                        0                2                     0   
...                    ...              ...                   ...   
2223                     1                3                     0   
2224                     0                0                     0   
2225                     1                3                     1   
2226                     0                2                     0   
2227                     0                2                     2   

      Full_Buy_Rounds  Full_Buy_Rounds_Wons  Round1_Spent  Round1_Result  \
0                  15                     6          4300          False   
1                  19                    14          4200           True   
2                  21                    13          4200           True   
3                  17                     8          4250          False   
4                  23                    16          4200          False   
...               ...                   ...           ...            ...   
2223                8                     1          4100           True   
2224               20                    15          3850          False   
2225               13                     4          4100           True   
2226               19                    11          4300          False   
2227               19                    11          4150           True   

      Round2_Spent  Round2_Result  Round3_Spent  Round3_Result  Round4_Spent  \
0             9400          False          1000          False         27950   
1            18750           True         22350           True         22200   
2            19750          False          9550          False         15000   
3             9650           True         22650           True         31650   
4            11900          False          1400          False         24350   
...            ...            ...           ...            ...           ...   
2223         19650          False          7150          False          1800   
2224          1500          False         21400           True         24300   
2225         19750           True         25550          False          3550   
2226          1200          False         20950          False         20500   
2227         18350           True         22750           True         27100   

      Round4_Result  Round5_Spent  Round5_Result  Round6_Spent  Round6_Result  \
0             False         11550          False         29250          False   
1              True         28450           True         26650           True   
2             False          6800          False         24600           True   
3              True         31

In [51]:
csgoeco_map_unique = csgo_match_economy['MapName'].nunique()
csgoeco_mid_unique = csgo_match_economy['MatchID'].nunique()

csgo_match_ecorow = len(csgo_match_economy)
csgo_meco_mergerow = len(csgo_meco_merge)
lost_merge_row3 = (csgo_match_ecorow-csgo_meco_mergerow)/csgo_match_ecorow*100

print(csgo_map_print)
print("CS:GO Match Economy have "+str(csgoeco_map_unique)+" unique 'MapName' values, and "+
      str(csgoeco_mid_unique)+" unique 'MatchID' values")
print("Before, CS:GO Match Economy has "+str(csgo_match_ecorow)+" rows. \
      \nCS:GO Match Economy now have "+str(csgo_meco_mergerow)+" rows.\
      \nCS:GO Match Economy lost "+str(round(lost_merge_row3, 2))+" rows after merge it with CS:GO Match Map based on column 'MatchID' and 'MapName'.")

CS:GO Match Map have 7 unique 'MapName' values, and 490 unique 'MatchID' values
CS:GO Match Economy have 8 unique 'MapName' values, and 9334 unique 'MatchID' values
Before, CS:GO Match Economy has 39914 rows.       
CS:GO Match Economy now have 2228 rows.      
CS:GO Match Economy lost 94.42 rows after merge it with CS:GO Match Map based on column 'MatchID' and 'MapName'.


#### CS:GO Match Economy Results
Merge CS:GO Match Economy with CS:GO Match Map based on column 'MatchID' and 'MapName' on both side
- Before 39914 rows
- After 2228 rows
- Lost 94.42 rows

Extract CS:GO Match Economy to Excel.

In [52]:
csgo_meco_merge.to_excel(('csgo_meco_merge.xlsx'))